metadata-ingestion/src/datahub/ingestion/source/mongodb.py

```diff
@@ -100,7 +100,7 @@ class MongoDBConfig(
     enableSchemaInference: bool = Field(
         default=True, description="Whether to infer schemas. "
     )
-    schemaSamplingSize: Optional[PositiveInt] = Field(
+    schemaSamplingSize: Optional[NonNegativeInt] = Field(
         default=1000,
         description="Number of documents to use when inferring schema size. If set to `0`, all documents will be scanned.",
```

Comment:
It seems like the current behavior will already do the right thing if schemaSamplingSize is set to null. I might be missing something, but it seems like it'd make more sense to simply fix the docs here and leave the actual code unchanged

```suggestion
        description="Number of documents to use when inferring schema size. If set to `null`, all documents will be scanned.",
```

Reply:
I've made a small update on the linked issue on my first message

> As far as I have reached, I think that it's imposible to force a full mongodb collection scan, because the yaml parser + validator forces to provide a value greater than 0, but internal code checks for null

I'm gonna try to explain my thoughts on detail:

1. `schemaSamplingSize: Optional[PositiveInt]` does allow `null` value, but not `0` value so I changed it to `NonNegativeInt`
2. `schemaSamplingSize` is also asigned to a Pydantic Field with the following default -> `default=1000,`. So, if no value is provided, this will have that default value
3. You can't  explicit provide a `null` or `none` value trough the yaml parser (as i also reflect now on the issue)

So, if I need to provide a value to the parser and it can't be null, providing a 0 value could be a good option.